crates/cxx-qt-lib/src/core/qbytearray.rs

```diff
@@ -122,6 +125,97 @@ impl AsRef<[u8]> for QByteArray {
     }
 }
 
+/// A QByteArray cursor allows seeking inside a QByteArray
+/// and writing to it.
+/// Comparable with std::io::Cursor.
+pub struct QByteArrayCursor {
```

Comment:
can we split the QByteArray changes into a separate commit?

Reply:
Done